include/aspect/volume_of_fluid/assembly.h

```diff
@@ -0,0 +1,220 @@
+    }
+  }
+
+  namespace Assemblers
+  {
+
+    /**
+     * Class to hold VolumeOfFluid assembly logic, as analogous to that used in the main simulator.
+     */
+    template <int dim>
+    class VolumeOfFluidAssembler : public SimulatorAccess<dim>
+    {
+      public:
+        /**
+         * Do setup and assembly on internal quadrature points and dispatch to
+         * other functions for face assembly
+         */
+        void local_assemble_volume_of_fluid_system (const VolumeOfFluidField<dim> field,
+                                                    const unsigned int calc_dir,
+                                                    const bool update_from_old,
+                                                    const typename DoFHandler<dim>::active_cell_iterator &cell,
+                                                    internal::Assembly::Scratch::VolumeOfFluidSystem<dim> &scratch,
+                                                    internal::Assembly::CopyData::VolumeOfFluidSystem<dim> &data) const;
+
+        /**
+         * Do assembly for cell faces on the boundary
+         */
+        void local_assemble_boundary_face_volume_of_fluid_system (const VolumeOfFluidField<dim> field,
+                                                                  const unsigned int calc_dir,
+                                                                  const bool update_from_old,
+                                                                  const typename DoFHandler<dim>::active_cell_iterator &cell,
+                                                                  const unsigned int face_no,
+                                                                  internal::Assembly::Scratch::VolumeOfFluidSystem<dim> &scratch,
+                                                                  internal::Assembly::CopyData::VolumeOfFluidSystem<dim> &data) const;
+
+        /**
+         * Function for assembling face fluxes for VolumeOfFluid system.
+         */
+        void local_assemble_internal_face_volume_of_fluid_system (const VolumeOfFluidField<dim> field,
+                                                                  const unsigned int calc_dir,
+                                                                  bool update_from_old,
+                                                                  const typename DoFHandler<dim>::active_cell_iterator &cell,
+                                                                  const unsigned int face_no,
+                                                                  internal::Assembly::Scratch::VolumeOfFluidSystem<dim> &scratch,
+                                                                  internal::Assembly::CopyData::VolumeOfFluidSystem<dim> &data) const;
+
+        /**
+         * Set volume fraction threshold for use in assembly
+         */
+        void set_volume_fraction_threshold(const double value);
+
+      private:
+        double volume_fraction_threshold;
```

Comment:
Couldnt you call `this->get_volume_of_fluid_handler().get_volume_fraction_threshold` instead of copying the information here? That would be safer.